insights/parsers/cpuinfo.py

```diff
@@ -143,18 +143,43 @@ def parse_content(self, content):
             "revision": "revision",
             "address sizes": "address_sizes",
             "bugs": "bugs",
-            "microcode": "microcode"
+            "microcode": "microcode",
+            "cpu MHz static": "clockspeeds",
+            "features": "features"
         }
 
         for line in get_active_lines(content, comment_char="COMMAND>"):
             key, value = [p.strip() for p in line.split(":", 1)]
+            # For s390x the : symbol is after the number instead of before.
+            # So re-split and set the key and value before checking mappings.
+            if key.startswith("processor") and key[-1].isdigit():
+                key, value = key.split(" ", 1)
+
             if key in mappings:
                 self.data[mappings[key]].append(value)
 
         if "cpu" in self.data and "POWER" in self.data["cpu"][0]:
             # this works differently than on x86 and is not per-cpu
+            model_id = self.data["model_ids"][0]
             del self.data["model_ids"]
```

Comment:
How about not `deleting` but just re-assigning it with the list of values (with the length of the number of `cpus`)?
```python
cpu_cnt = len(self.data['cpus'])
...
self.data["model_ids"]  = [model_id] * cpu_cnt
```

Reply:
@xiangce Thanks for the suggestion I didn't think about that, I updated the commit with your suggestions.